atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableTimestampsReader.java

```diff
@@ -48,52 +46,63 @@
+            Optional<Long> candidateFine = getCandidatesInCoarsePartition(shardAndStrategy, current, range.get());
             if (candidateFine.isPresent()) {
                 return candidateFine;
             }
-            currentCoarse++;
+            current++;
         }
         return Optional.empty();
     }
 
-    List<Long> getCandidatesInCoarsePartition(ShardAndStrategy shardAndStrategy, long partitionCoarse) {
-        byte[] row = SweepableTimestampsTable.SweepableTimestampsRow.of(
-                shardAndStrategy.shard(),
-                partitionCoarse,
-                PersistableBoolean.of(shardAndStrategy.isConservative()).persistToBytes())
+    private Optional<BatchColumnRangeSelection> getColRangeSelection(long minFine, long maxFine) {
+        if (minFine >= maxFine) {
+            return Optional.empty();
+        }
+        byte[] start = SweepableTimestampsTable.SweepableTimestampsColumn.of(minFine).persistToBytes();
+        byte[] end = SweepableTimestampsTable.SweepableTimestampsColumn.of(maxFine).persistToBytes();
+        return Optional.of(BatchColumnRangeSelection.create(start, end, 1));
+    }
+
+    Optional<Long> getCandidatesInCoarsePartition(ShardAndStrategy shardAndStrategy, long partitionCoarse,
+            BatchColumnRangeSelection colRange) {
+        byte[] row = SweepableTimestampsTable.SweepableTimestampsRow.of(shardAndStrategy.shard(),
+                    partitionCoarse,
+                    PersistableBoolean.of(shardAndStrategy.isConservative()).persistToBytes())
                 .persistToBytes();
-        RangeRequest request = RangeRequest.builder()
-                .startRowInclusive(row)
-                .endRowExclusive(RangeRequests.nextLexicographicName(row))
-                .retainColumns(ColumnSelection.all())
-                .batchHint(1)
-                .build();
-        ClosableIterator<RowResult<Value>> rowResultIterator = kvs.getRange(TABLE_REF, request, SweepQueueUtils.CAS_TS);
-
-        if (!rowResultIterator.hasNext()) {
-            return ImmutableList.of();
+
+        Map<byte[], RowColumnRangeIterator> test = kvs.getRowsColumnRange(
+                TABLE_REF, ImmutableList.of(row), colRange, SweepQueueUtils.CAS_TS);
+
+        RowColumnRangeIterator col = Iterables.getOnlyElement(test.values());
+        if (!col.hasNext()) {
+            return Optional.empty();
         }
+        Map.Entry<Cell, Value> firstColumnEntry = col.next();
 
-        RowResult<Value> rowResult = rowResultIterator.next();
+        return Optional.of(getPartitionFromEntry(firstColumnEntry));
+    }
 
-        return rowResult.getColumns().keySet().stream()
-                .map(SweepableTimestampsTable.SweepableTimestampsColumn.BYTES_HYDRATOR::hydrateFromBytes)
-                .map(SweepableTimestampsTable.SweepableTimestampsColumn.getTimestampModulusFun())
-                .collect(Collectors.toList());
+    private long getPartitionFromEntry(Map.Entry<Cell, Value> entry) {
```

Comment:
Might as well pass `col` from above into here, and do the plumbing in one method.